Sources/GRPCHTTP2Core/Server/HTTP2ServerTransport.swift

```diff
@@ -188,4 +188,135 @@ extension HTTP2ServerTransport.Config {
+    package var tlsConfig: TLS? {
+      switch wrapped {
+      case .plaintext:
+        return nil
+      case .tls(let config):
+        return config
+      }
+    }
+  }
+
+  public struct TLS: Sendable {
+    /// The serialization format of the provided certificates and private keys.
+    public struct SerializationFormat: Sendable, Equatable {
+      package enum Wrapped {
+        case pem
+        case der
+      }
+
+      package let wrapped: Wrapped
+
+      public static let pem = Self(wrapped: .pem)
+      public static let der = Self(wrapped: .der)
+    }
+
+    public struct CertificateSource: Sendable {
+      package enum Wrapped {
+        case file(path: String, format: SerializationFormat)
+        case bytes(bytes: [UInt8], format: SerializationFormat)
+      }
+
+      package let wrapped: Wrapped
+
+      /// The certificate will be provided via a file.
+      public static func file(path: String, format: SerializationFormat) -> Self {
+        Self(wrapped: .file(path: path, format: format))
+      }
+
+      /// The certificate will be provided as an array of bytes.
+      public static func bytes(_ bytes: [UInt8], format: SerializationFormat) -> Self {
+        Self(wrapped: .bytes(bytes: bytes, format: format))
+      }
+    }
+
+    public struct PrivateKeySource: Sendable {
+      package enum Wrapped {
+        case file(path: String, format: SerializationFormat)
+        case bytes(bytes: [UInt8], format: SerializationFormat)
+      }
+
+      package let wrapped: Wrapped
+
+      /// The private key will be provided via a file.
+      public static func file(path: String, format: SerializationFormat) -> Self {
+        Self(wrapped: .file(path: path, format: format))
+      }
+
+      /// The private key will be provided as an array of bytes.
+      public static func bytes(
+        _ bytes: [UInt8],
+        format: SerializationFormat
+      ) -> Self {
+        Self(wrapped: .bytes(bytes: bytes, format: format))
+      }
+    }
+
+    public struct TrustRootsSource: Sendable {
+      package enum Wrapped {
+        case certificates([CertificateSource])
+        case systemDefault
+      }
+
+      package let wrapped: Wrapped
+
+      /// Path to either a file of CA certificates in PEM format, or a directory containing CA certificates in PEM format.
+      public static func certificates(_ certificateSources: [CertificateSource]) -> Self {
+        Self(wrapped: .certificates(certificateSources))
+      }
+
+      /// The system default trust.
+      public static let systemDefault: Self = Self(wrapped: .systemDefault)
+    }
+
+    /// The certificate the server will offer during negotiation.
+    public var certificateChainSources: [CertificateSource]
+
+    /// The private key associated with the leaf certificate.
+    public var privateKeySource: PrivateKeySource
+
+    /// Whether to verify the remote certificate.
+    public var verifyClientCertificate: Bool
```

Comment:
Thinking about this one a little more, I wonder if we should expose it at all, or, if we do we might need more control over it because for mTLS we don't always want full verification, we want no hostname verification which can't do right now.

Reply:
We could expose an enum with all 3 options (basically how NIOSSL exposes it) instead?